Go attempt (#3)
ParamPatel207/app_action: README.md

> # Deploy a [DigitalOcean App Platform](https://www.digitalocean.com/products/app-platform/) app using GitHub Actions.

Reviewer: Do we need a disclaimer that there app has to already exist?

Author: How about?
# Redeploy a [DigitalOcean App Platform](https://www.digitalocean.com/products/app-platform/) app using GitHub Actions.

Author: i think above title can imply that app should be deployed without disclaimer

Reviewer: I think it'd be confusing. Let's leave it as is and see if it becomes a problem.

 - Auto-deploy your app from source on commit, while allowing you to run tests or perform other operations before.
 - Auto-deploy your app from source and also update DigitalOcean Container Registry (DOCR) configuration in DigitalOcean [App Spec](https://docs.digitalocean.com/products/app-platform/references/app-specification-reference/) and deploy application with updated DOCR image.

**Note: This action only supports DOCR configuration changes for Auto-deploy**

# Usage

> ### DigitalOcean App Platform Auto-deploy with same app spec.

Reviewer: ```suggestion
### Deploy via GH Action and let DigitalOcean App Platform build and deploy your app.
```

- Get DigitalOcean Personal Access token by following this [instructions](https://docs.digitalocean.com/reference/api/create-personal-access-token/).**(skip this step if you already have DigitalOcean Personal Access Token)**
- Declare DigitalOcean Personal Access Token as DIGITALOCEAN_ACCESS_TOKEN variable in the [secrets](https://docs.github.com/en/actions/reference/encrypted-secrets#creating-encrypted-secrets-for-a-repository) of github repository. 

> - Add this step to deploy your application on DigitalOcean App Platform without changing any app spec configuration or making any other changes.

Reviewer: ```suggestion
- [Create a GitHub Action workflow file](https://docs.github.com/en/actions/learn-github-actions/introduction-to-github-actions#create-an-example-workflow) and add this step below to it or add this to your existing action.
```

  ```

> - DigitalOcean App Platform will now deploy your application.

Reviewer: ```suggestion
- This step will trigger a deploy to your App on DigitalOcean App Platform
```

Author: updated using Commit suggestion option.

> ### Update DigitalOcean Container Registry(DOCR) configuration of multiple component in App Spec

Reviewer: ```suggestion
### Deploy an one or more app components from a DigitalOcean Container Registry (DOCR) 
```

Author: updated using commit suggestion

>                       ]'

Reviewer: Add some example data

Author: I intentionally skipped the example data so that user can copy the format and paste it in their workflow. Because while doing research for headers i found that while reading documentation many users prefer to copy from documentation prompt.

  ```

> - DigitalOcean App Platform will now update your DOCR information in App Spec and then deploy your application.

Reviewer: ```suggestion
- DigitalOcean App Platform will now update your DOCR information in App Spec and then deploy your application.
- This step will trigger a DigitalOcean App Platform deployment of your app using the images specified.
```

> **Note: Always use unique tag names to push image to the DigitalOcean Container Registry. This will allow you to deploy your application without delay. [ref](https://docs.digitalocean.com/products/container-registry/quickstart/)**

Reviewer: Give an example, eg using git short hashes

Author: git short hashes is shown in example https://github.com/ParamPatel207/docr_sample. Should I use

- name: Install doctl
  uses: digitalocean/action-doctl@v2
  with:
  token: ${{ secrets.DIGITALOCEAN_ACCESS_TOKEN }}
- name: Generate GITHUB_SHA
  id: github-sha
  shell: bash
  run: |
    SHORT_SHA=$(echo $GITHUB_SHA | cut -c1-7)
    echo "::set-output name=sha::$SHORT_SHA"
- name: Publish Image to Digital Ocean Container Registry
  shell: bash
  run: |
    doctl registry login --expiry-seconds 600
    docker build . -t registry.digitalocean.com/sample-go/add_sample:${{steps.github-sha.outputs.sha}}
    docker push registry.digitalocean.com/sample-go/add_sample:${{steps.github-sha.outputs.sha}}
   
as an example for short sha and push to docker??????

# Inputs
- `app_name` - Name of the app on App Platform.

> - `list_of_image` - (optional)List of json object for providing information about name, repository and tag of the image in docr.(by default latest tag is used)

Reviewer: can we rename this to `images`?

Reviewer: also we advise against using `latest` should we use the commit short sha by default?

Author: I believe user should have option to declare their own tags. For example; they may prefer to tag images by version number which might be unique.
Saying that we can provide an optional feature to deploy using sha tag but as users are deploying their docker images to docr by themselves we might need to provide handling of deployment from our side. For example; We take same input except tag field and then deploy dockerfile to docr with generated sha tag and then update our app platform app spec.

Reviewer: I'm suggesting we change the default from `latest` which has known issues to something that doesn't and is common across all projects. They're still welcome to set whatever they want but we can give them a nice default that is going to work without issue.

Author: Actually they are the one pushing the images and we are the one pulling so what i am trying to say is they need to push image using sha hash. The default is latest because if the user does not mention tag DOCR uses latest as tag. If we set sha hash as default it might confuse users as now they have to explicitly set sha key and also push image using same key. This might be good in regulating the flow but I believe there may be many users who are not using tags for docr upload. So for example; I am user writing github action and I forgot to tag the image our action will be trying to retrieve tag based on hash rather then using latest.

Reviewer: Aaah I see, you're right. I misunderstood! thanks for explaining 👍
